lms/templates/main.html

```diff
@@ -165,7 +165,9 @@
 </html>
 
 <%def name="login_query()">${
-  u"?next={0}".format(urlquote_plus(login_redirect_url)) if login_redirect_url else ""
+  u"?next={next}".format(
```

Comment:
Why does it need to be bokchoy?  Can't you just validate that the rendered template has the appropriate url under the various conditions?